extensions/hibernate-reactive/deployment/src/main/java/io/quarkus/hibernate/reactive/deployment/HibernateReactiveProcessor.java

```diff
@@ -101,6 +102,13 @@ void reflections(BuildProducer<ReflectiveClassBuildItem> reflectiveClass) {
         reflectiveClass.produce(new ReflectiveClassBuildItem(false, false, REFLECTIVE_CONSTRUCTORS_NEEDED));
     }
 
+    @BuildStep
+    void services(BuildProducer<ServiceProviderBuildItem> producer) {
+        producer.produce(
+                new ServiceProviderBuildItem(org.hibernate.service.spi.SessionFactoryServiceContributor.class.getName(),
+                        org.hibernate.reactive.service.internal.ReactiveSessionFactoryServiceContributor.class.getName()));
```

Comment:
Services are more @Sanne expertise, but it seems to make sense.
I'm surprised that the failure only happens for native queries if this is what was missing.

Reply:
`ServiceProviderBuildItem` is only needed for native as the ServiceLoader needs explicit configuration in that case (as opposed to JVM mode where it just works based on the what's on the classpath)

Reply:
I'm a bit surprised that this approach works at all, `SessionFactoryServiceContributor` should be disabled as the preferred way to registering services in Quarkus is a bit different.

But if it works we can certainly merge.. fixes get priority. I'll just make a note to investigate this further.

Reply:
I also checked the reproducer and it does work.
Obviously I'm not the SME here, so it's up to you to decide 😎